stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -1,24 +1,33 @@
-import React, { useEffect, useState, useCallback } from "react";
+import React, { useEffect, useState, useCallback, useReducer } from "react";
 import Components from "stockflux-components";
 import { Link } from "react-router-dom";
 import "./SecuritiesTable.css";
 import * as service from "../services/SecuritiesService";
 import ValidationError from "../services/ValidationError";
-import Alert from "./Alert";
+import Alert, { AlertType } from "./Alert";
 import { TableState } from "../enums";
-import Button from "./Button";
+import ToolTip from "./ToolTip";
+import Button, { ButtonSize } from "./Button";
+import { FaPen, FaTrashAlt, FaEye, FaEyeSlash } from "react-icons/fa";
+import { MdClose, MdCheck } from "react-icons/md";
+import {
+  securitiesTableReducer,
+  initialTableState
+} from "../reducers/securitiesTableReducer";
 
 const SecuritiesTable = ({ location }) => {
   const [securitiesData, setSecuritiesData] = useState([]);
-  const [messages, setMessages] = useState([]);
-  const [state, setState] = useState(TableState.LOADING);
+  const [state, dispatch] = useReducer(
+    securitiesTableReducer,
+    initialTableState
+  );
 
   const handleError = err => {
-    setState(TableState.ERROR);
+    dispatch({ type: "error" });
```

Comment:
I think I've commented on this already but this is redundant. You are already going to dispatch error in either if or else below.

Reply:
Good spot, I don't think you mentioned that earlier but have now removed it

> InputForm.js, SecuritiesTable.js are still missing proptypes.

I've added propTypes to these components now